src/types/package.rs

```diff
@@ -450,6 +482,107 @@ pub struct PackageDescriptor {
     pub package_type: PackageType,
 }
 
+/// Describes a package descriptor in the system with lockfile path
+#[derive(
+    PartialEq, Eq, PartialOrd, Ord, Hash, Clone, Debug, Serialize, Deserialize, JsonSchema,
+)]
+pub struct LockFilePackageDescriptor {
+    pub package_descriptor: PackageDescriptor,
+    pub lockfile_path: Option<String>,
+}
+
+impl From<&PackageDescriptor> for LockFilePackageDescriptor {
+    fn from(value: &PackageDescriptor) -> Self {
+        LockFilePackageDescriptor {
+            package_descriptor: value.clone(),
+            lockfile_path: None,
+        }
+    }
+}
+
+/// Describes a package specifier in the system with lockfile path
+#[derive(
+    PartialEq, Eq, PartialOrd, Ord, Hash, Clone, Debug, Serialize, Deserialize, JsonSchema,
+)]
+pub struct LockFilePackageSpecifier {
+    pub package_specifier: PackageSpecifier,
+    pub lockfile_path: Option<String>,
+}
+
+impl From<&PackageSpecifier> for LockFilePackageSpecifier {
+    fn from(value: &PackageSpecifier) -> Self {
+        LockFilePackageSpecifier {
+            package_specifier: value.clone(),
+            lockfile_path: None,
+        }
+    }
+}
+
+/// Describes a package url in the system with lockfile path
+#[derive(
+    PartialEq, Eq, PartialOrd, Ord, Hash, Clone, Debug, Serialize, Deserialize, JsonSchema,
+)]
+pub struct ParsedPurl {
+    pub purl: String,
+    pub lockfile_path: Option<String>,
+}
+
+impl TryFrom<&PackageDescriptor> for ParsedPurl {
+    type Error = purl::PackageError;
+
+    fn try_from(value: &PackageDescriptor) -> Result<Self, Self::Error> {
+        let package_type = purl::PackageType::from(value.package_type);
+        let purl = Purl::builder(package_type, &value.name)
+            .with_version(&value.version)
+            .build()?
+            .to_string();
```

Comment:
This isn't right and it's going to fail for all maven packages, most go packages, and many npm packages. You can copy (or maybe even move) from `package_url.rs` `purl_from_parts`.

Reply:
Ah yes, thanks for pointing that out. I was updating tests last night and ran into some failures and noticed you had documented failures regarding maven in the crate.